x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -882,4 +889,18 @@ private static void checkCastOnInexact(LogicalPlan p, Set<Failure> localFailures
             }
         }, Cast.class)), Filter.class);
     }
+
+    private static void checkClientSupportsDataTypes(LogicalPlan p, Set<Failure> localFailures, SqlVersion version) {
```

Comment:
I think it's better to just take a look at the output of the plan, regardless of whether a select is used or not. That is you don't have to analyze the structure only its output.

Reply:
Thanks, changed.